factory/stateComponents.go

```diff
@@ -147,36 +147,28 @@ func (scf *stateComponentsFactory) createAccountsAdapters(triesContainer common.
 		ProcessingMode:        scf.processingMode,
 		ProcessStatusHandler:  scf.core.ProcessStatusHandler(),
 	}
-	accountsAdapterAPI, err := state.NewAccountsDB(argsAPIAccountsDB)
-	if err != nil {
-		return nil, nil, nil, fmt.Errorf("accounts adapter API: %w: %s", errors.ErrAccountsAdapterCreation, err.Error())
-	}
 
-	wrapper, err := state.NewAccountsDBApi(accountsAdapterAPI, scf.chainHandler)
+	accountsAdapterApiOnFinal, err := factoryState.CreateAccountsAdapterAPIOnFinal(argsAPIAccountsDB, scf.chainHandler)
 	if err != nil {
-		return nil, nil, nil, fmt.Errorf("accounts adapter API: %w: %s", errors.ErrAccountsAdapterCreation, err.Error())
+		return nil, nil, nil, fmt.Errorf("accounts adapter API on final: %w: %s", errors.ErrAccountsAdapterCreation, err.Error())
 	}
 
-	accountsAdapterForRepositoryOnFinal, err := state.NewAccountsDB(argsAPIAccountsDB)
+	accountsAdapterApiOnCurrent, err := factoryState.CreateAccountsAdapterAPIOnCurrent(argsAPIAccountsDB, scf.chainHandler)
```

Comment:
Oh, I see now, before I was passing an accounts DB - that would have been perfectly safe in the first implementation - until someone would have accessed `innerAccountsAdapter.Commit()` or so, right?